### PR TITLE
Docs rewrite

### PR DESCRIPTION
**Pull request by JustaSqu1d**
## Summary

This pull request fixes many broken references and grammar. It also documents Auto Moderation and Forum Channels.

## Information

- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

**Comment by Lulalaby** [comment]
Thanks squid, as always <3

**Review by Lulalaby — DISMISSED** [review]

File: docs/api.rst [hunk 1]
```diff
@@ -469,7 +469,7 @@ to handle it, which defaults to print a traceback and ignoring the exception.
 
 .. function:: on_socket_event_type(event_type)
 
-    Called whenever a websocket event is received from the WebSocket.
+    Called whenever a WebSocket event is received from the WebSocket.
```
Comment (Dorukyum) [inline docs/api.rst:472]: Do we have to change this?

**Review by Dorukyum — COMMENTED** [review]
That was a lot of merge conflicts.

File: docs/api.rst (hunk 1 above)
Comment (JustaSqu1d) [reply docs/api.rst:472]: Not necessarily but grammatically, yes.
image

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]
*worry*

**Pull request merged**